### PR TITLE
Deprecate `none()` step in favour of `discard()`

### PR DESCRIPTION
**Pull request by ryn5**
Deprecates `none()`, which is primarily used by `iterate()` to discard traversal results in remote contexts, in favour of `discard`.  As this is a breaking change, `none()` will be removed at a later date in the next major release.  

UPDATE: Bytecode for discard has been reverted to use "none" to avoid incompatibility issues if a 3.7.2+ driver is interacting with a 3.7.1 server.  This will be updated with the removal of `none()` in https://github.com/apache/tinkerpop/pull/2377.

UPDATE: Introduction of discard has been pushed to 4.0.0.  This PR will only be deprecating none for 3.7.2.

UPDATE: None will simply be renamed to discard in 4.0.0 without deprecation.

### CODE REVIEW

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/__.java
Comment (ryn5) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/__.java:1]: Unsure whether this is correct, but solves the issue of the shouldHaveMethodsOfGraphTravesralOnAnonymousGraphTraversal test failing with error message "__ is missing the following method: discard:[]" in GraphTraversalTest.java.

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -55,6 +55,7 @@ This release also includes changes from <<release-3-6-6, 3.6.6>> and <<release-3
 * Fixed bug in `union()` as a start step where the `Path` was including the starting dummy traverser.
 * Moved some TinkerGraph specific transaction tests from `TransactionMultiThreadedTest` to `TinkerTransactionGraphTest`
 * Fixed incorrect read operations in some cases for `TinkerTransactionGraph`.
+* Deprecated `none()` in favor of  `discard()` to make room for a list filtering `none()` step.
```
Comment (spmallette) [inline CHANGELOG.asciidoc:58]: this should be moved to 3.7.2 section

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -82,7 +83,7 @@ This release also includes changes from <<release-3-6-6, 3.6.6>> and <<release-3
 * TINKERPOP-2830 Handle User-Agent from HTTP Requests to server
 * TINKERPOP-2946 Resolve ordering issues in gherkin tests
 * TINKERPOP-2951 Add translator to the Go GLV
-* TINKERPOP-2964  Many TraversalParent's steps have a replaceLocalChild logic that can result in a new ChildTraversal having an ID that already exists.
+* TINKERPOP-2964 Many TraversalParent's steps have a replaceLocalChild logic that can result in a new ChildTraversal having an ID that already exists.
```
Comment (spmallette) [inline CHANGELOG.asciidoc:86]: we don't directly add JIRA to CHANGELOG - they are added on release.

File: docs/src/upgrade/release-3.7.x.asciidoc [hunk 3]
```diff
@@ -33,6 +33,9 @@ complete list of all the modifications that are part of this release.
 
 === Upgrading for Providers
 
+==== Renaming None Step to Discard
```
Comment (spmallette) [inline docs/src/upgrade/release-3.7.x.asciidoc:36]: i suggest "Deprecated none() step" as a title so that "deprecate" is clearly there in the toc.

Comment (ryn5) [reply docs/src/upgrade/release-3.7.x.asciidoc:36]: Will do, thanks!

File: CHANGELOG.asciidoc (hunk 2 above)
Comment (spmallette) [reply CHANGELOG.asciidoc:86]: i'll file this under "Ways to show folks you're not actually reading anything in the code review." haha

File: docs/src/upgrade/release-3.7.x.asciidoc (hunk 3 above)
Comment (ryn5) [reply docs/src/upgrade/release-3.7.x.asciidoc:39]: Updated providers section and added user upgrade section

**Comment by ryn5** [comment]
Reverted all changes to simply provide notice of the upcoming renaming in https://github.com/apache/tinkerpop/pull/2408.

**Pull request closed without merging**